client/my-sites/themes/controller/index.node.js

```diff
@@ -43,27 +38,8 @@ export function fetchThemeDetailsData( context, next ) {
 	}
 
 	const themeSlug = context.params.slug;
-	const theme = themeDetailsCache.get( themeSlug );
-
-	if ( theme ) {
-		debug( 'found theme!', theme.id );
-		context.store.dispatch( receiveThemeDetails( theme ) );
-		next();
-	}
 
-	themeSlug && wpcom.undocumented().themeDetails( themeSlug, ( error, data ) => {
-		if ( error ) {
-			debug( `Error fetching theme ${ themeSlug } details: `, error.message || error );
-			return;
-		}
-		const themeData = themeDetailsCache.get( themeSlug );
-		if ( ! themeData || ( Date( data.date_updated ) > Date( themeData.date_updated ) ) ) {
-			debug( 'caching', themeSlug );
-			themeDetailsCache.set( themeSlug, data );
-			context.store.dispatch( receiveThemeDetails( data ) );
-			next();
-		}
-	} );
+	themeSlug && context.store.dispatch( fetchThemeDetails( themeSlug, next ) );
```

Comment:
Yeah, that's right. So if we want to avoid that, we could re-enable some sort of caching. As I said, my preferred approach would be encapsulating it in our Redux tree. We could introduce a `getThemeDetails()` selector and use that in lieu of L46-L52. The reducer is already storing theme details per-theme-id IIRC, so it might work OOTBish...